File: js/009 Palindrome Number/PalindromeNumber.js

/**
 * 本题题意：判斷回文數字
 * @param {number} x
 * @return {boolean}
 */
var isPalindrome = function(x) {
  for (let i = 0, str = x + "", n = parseInt(str.length / 2); i < n; i++) {
    if (str[i] != str[str.length - 1 - i]) {
      return false;
    }
  }
  return true;
};

File: go/009 Palindrome Number/PalindromeNumber.go
package main
/*
* 本题题意：判斷回文數字
*/

import "strconv"
func isPalindrome(x int) bool {
  str := strconv.Itoa(x)
  length := len(str)
  for i, n := 0, length / 2; i < n; i++ {
    if str[i] != str[length - 1 - i] {
      return false;
    }
  }
  return true;
}
